logbook-spring-boot-autoconfigure/src/test/java/org/zalando/logbook/autoconfigure/IncludeTest.java

```diff
@@ -21,89 +23,176 @@
+
+            assertThat(logsList).hasSize(1);
+        }
 
-        assertThat(logsList).isEmpty();
     }
 
-    @Test
-    void shouldExcludeInternalApiWithQueryParameters() throws IOException {
-        logbook.process(request("/internal-api").withQuery("debug=true")).write();
+    @Nested
+    @LogbookTest(imports = IncludeTest.HeaderConfig.class)
+    class IncludeHeaderTest {
+
+        @Autowired
+        private Logbook logbook;
+
+        @MockBean
+        private HttpLogWriter writer;
+
+        @BeforeEach
+        void setUp() {
+            doReturn(true).when(writer).isActive();
+        }
+
+
+        @ParameterizedTest
+        @CsvSource(value = {
+                "'host', 'localhost', true",
+                "'host', '127.0.0.1', false",
+                "'host', '', false",
+                "'localhost', 'localhost', false",
+        })
+        void shouldIncludeExpectedRequests(String key, String value, boolean shouldLog) throws IOException {
+            logbook.process(request("/api").withHeaders(HttpHeaders.of(key, value))).write();
+
+            VerificationMode verificationMode = shouldLog ? atLeastOnce() : never();
+
+            verify(writer, verificationMode).write(any(Precorrelation.class), any());
+        }
 
-        assertThat(logsList).isEmpty();
     }
 
-    @Test
-    void shouldNotExcludeApiWithParameter() throws IOException {
-        logbook.process(request("/api").withQuery("debug=true")).write();
+    @Nested
+    @LogbookTest(imports = IncludeTest.WildcardHeaderConfig.class)
+    class IncludeWildcardHeaderTest {
+
+        @Autowired
+        private Logbook logbook;
+
+        @MockBean
+        private HttpLogWriter writer;
+
+        @BeforeEach
+        void setUp() {
+            doReturn(true).when(writer).isActive();
+        }
+
+
+        @ParameterizedTest
+        @CsvSource(value = {
+                "'host', 'localhost', true",
+                "'host', '127.0.0.1', true",
+                "'host', '', true",
+                "'hosts', 'localhost', false",
+                "'hosts', '', false",
+        })
+        void shouldIncludeExpectedRequests(String key, String value, boolean shouldLog) throws IOException {
+            logbook.process(request("/api").withHeaders(HttpHeaders.of(key, value))).write();
+
+            VerificationMode verificationMode = shouldLog ? atLeastOnce() : never();
+
+            verify(writer, verificationMode).write(any(Precorrelation.class), any());
+        }
 
-        assertThat(logsList).hasSize(1);
     }
 
     private MockHttpRequest request(final String path) {
         return MockHttpRequest.create().withPath(path);
     }
 
+    @TestConfiguration
+    public static class PathConfig {
+        @Bean
+        public Predicate<HttpRequest> condition() {
+            return exclude(requestTo("/api/admin/**"));
+        }
+    }
+
+    @TestConfiguration
+    public static class HeaderConfig {
```

Comment:
It would be also good to use test `application-exclude.yml` and `application-include.yml` files for the tests instead of defining the predicate beans. This would test that the autoconfiguration logic that you adjusted `convertToPredicate` method works as expected. It will also work as a good example how this configuration can look like together with the rest of predicates. 

The `PathConfig` was a way to test how custom predicates work in a combination with the ones from .yml ones.

Reply:
Speaking about the test configuration in .yaml files. Would be nice to have examples of purely header predicates as well as the combination with other predicates. 
E.g. 
```
logbook:
  include:
  - /api/**
  predicate:
    include:
      - path: /another-api
      - path: /yet-another-api
        methods:
          - DELETE
      - methods:
          - PUT
      - path: /secured-api
        headers: 
           authorization: *
      - path: /only-gets-with-json
        method: GET
        headers: 
           accept: application/json
```